draft-ietf-quic-recovery.md

```diff
@@ -526,12 +526,11 @@ as a probe, unless there is no data available to send.  An endpoint MAY send up
 to two full-sized datagrams containing ack-eliciting packets, to avoid an
 expensive consecutive PTO expiration due to a single lost datagram.
 
-It is possible that the sender has no new or previously-sent data to send.  As
-an example, consider the following sequence of events: new application data is
-sent in a STREAM frame, deemed lost, then retransmitted in a new packet, and
-then the original transmission is acknowledged.  In the absence of any new
-application data, a PTO timer expiration now would find the sender with no new
-or previously-sent data to send.
+When the PTO timer expires, and there is new or previously sent data, it MUST
+be sent.  It is possible the sender has no new or previously-sent data to send.
+As an example, consider the following sequence of events: new application data
+is sent in a STREAM frame, deemed lost, then retransmitted in a new packet,
+and then the original transmission is acknowledged.
```

Comment:
This example only applies to implementations that can usefully process acknowledgements for packets they've already deemed lost and retransmitted the contents of, which I don't think is required. Is there a more common case?

Reply:
This is a fairly common case in our implementation, so I think it's a good example.

Reply:
Maybe note that this is optional behavior? I know if I was reading this for the first time, I'd try to work out where this was required and get confused.

Reply:
It probably should be recommended behavior, and we should probably have some text on it.  Filed #3058 to discuss if and what to do to improve that.